src/scenes/Events/EventEditor/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Drawer from '@material-ui/core/Drawer';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+
+import { closeEventEditor } from '../../../services/ui/actions';
+import { updateEvent, createEvent } from '../../../services/events/actions';
+
+const emptyEvent = {
+  name: '',
+  time: '',
+  desc: '',
+  id: -1,
+};
+
+class EventEditor extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      event: emptyEvent,
+      isNew: true,
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    // If prop hasn't change, don't edit state
+    if((props.event === null && state.event.id === -1) || (props.event && (props.event.id === state.event.id))) {
+      return state;
+    }
+    const newState =  {
+      event: props.event ? props.event : emptyEvent,
+      isNew: !props.event,
+    };
+    return newState;
+  }
+
+  handleChange(e, field) {
```

Comment:
This would be cleaner if it returned a function and used js clojures.

Ex.
```
handleSubmit(field) {
  return (e) => {
     ...
  };
}
```

That way onChange can just be `onChange={handleChange("fieldname")}`